### PR TITLE
Expressions tweaks

### PR DESCRIPTION
**Pull request by jasonLaster**
While I was looking at Expressions I found somethings to tweak:

* Errors should be shown
* Reference Errors should be `(unavailable)`

### CODE REVIEW

**Comment by jasonLaster** [comment]
this also fixes the keys issue

**Comment by jasonLaster** [comment]
this also fixes the duplication error: #3598

File: src/actions/expressions.js [hunk 1]
```diff
@@ -83,26 +79,23 @@ export function deleteExpression(expression: Expression) {
...
 
-function evaluateExpression(expression: Expression, frameId: frameIdType) {
+function evaluateExpression(expression: Expression) {
   return function({ dispatch, getState, client }: ThunkArgs) {
     if (!expression.input) {
       console.warn("Expressions should not be empty");
       return;
     }
 
+    const frameId = getSelectedFrameId(getState());
```
Comment (jasonLaster) [inline src/actions/expressions.js]: we don't need to pass in a frameId because we should always fetch it here

File: src/actions/pause.js [hunk 2]
```diff
@@ -231,18 +231,15 @@ export function breakOnNext() {
...
     );
 
-    const scopes = await client.getFrameScopes(frame);
-
-    dispatch({
-      type: "SELECT_FRAME",
-      frame,
-      scopes
-    });
+    dispatch(evaluateExpressions());
```
Comment (jasonLaster) [inline src/actions/pause.js]: we should evaluate the expressions after we select the frame

File: src/components/SecondaryPanes/Expressions.js [hunk 3]
```diff
@@ -13,42 +14,6 @@ import "./Expressions.css";
 import type { List } from "immutable";
 import type { Expression } from "../../types";
 
-function getValue(expression) {
```
Comment (jasonLaster) [inline src/components/SecondaryPanes/Expressions.js]: lets move this to a util where we can test it

File: src/reducers/pause.js [hunk 4]
```diff
@@ -229,9 +229,12 @@ export function getFrameScopes(state: OuterState, frameId: string) {
...
 
-const getSelectedFrameId = createSelector(getPauseState, pauseWrapper => {
-  return pauseWrapper.selectedFrameId;
-});
+export const getSelectedFrameId = createSelector(
+  getPauseState,
+  pauseWrapper => {
+    return pauseWrapper.selectedFrameId;
+  }
+);
```
Comment (jasonLaster) [inline src/reducers/pause.js]: isn't it nice to export just what you need

File: flow-typed/debugger-html.js [hunk 5]
```diff
@@ -178,8 +178,9 @@ declare module "debugger-html" {
  * @static
  */
   declare type Expression = {
-    id: number,
-    input: string
+    input: string,
```
Comment (jasonLaster) [inline flow-typed/debugger-html.js]: we weren't using id

File: src/actions/expressions.js [hunk 6]
```diff
@@ -24,22 +26,16 @@ export function addExpression(input: string) {
 
     const expression = getExpression(getState(), input);
     if (expression) {
-      return dispatch({
-        type: "UPDATE_EXPRESSION",
-        expression,
-        input
-      });
+      return dispatch(evaluateExpression(expression));
```
Comment (bomsy) [inline src/actions/expressions.js]: 👍

**Comment by bomsy** [comment]
Loooks nice! will test in a bit!

**Pull request merged**